pkg/tsdb/influxdb/influxql/response_parser.go

```diff
@@ -138,55 +147,34 @@ func newFrameWithTimeField(row models.Row, column string, colIndex int, query mo
 		}
 	}
 
-	timeField := data.NewField("Time", nil, timeArray)
+	timeField := data.NewField(timeColumnName, nil, timeArray)
 
 	var valueField *data.Field
 
 	switch valType {
 	case "string":
-		valueField = data.NewField("Value", row.Tags, stringArray)
+		valueField = data.NewField(valueColumnName, row.Tags, stringArray)
 	case "json.Number":
-		valueField = data.NewField("Value", row.Tags, floatArray)
+		valueField = data.NewField(valueColumnName, row.Tags, floatArray)
 	case "bool":
-		valueField = data.NewField("Value", row.Tags, boolArray)
+		valueField = data.NewField(valueColumnName, row.Tags, boolArray)
 	case "null":
-		valueField = data.NewField("Value", row.Tags, floatArray)
+		valueField = data.NewField(valueColumnName, row.Tags, floatArray)
 	}
 
 	name := string(formatFrameName(row, column, query, frameName[:]))
 	valueField.SetConfig(&data.FieldConfig{DisplayNameFromDS: name})
 	return newDataFrame(name, query.RawQuery, timeField, valueField)
 }
 
-func newFrameWithoutTimeField(row models.Row, retentionPolicyQuery bool, tagValuesQuery bool) *data.Frame {
+func newFrameWithoutTimeField(row models.Row, query models.Query) *data.Frame {
 	var values []string
 
-	if retentionPolicyQuery {
-		values = make([]string, 1, len(row.Values))
-	} else {
-		values = make([]string, 0, len(row.Values))
-	}
-
 	for _, valuePair := range row.Values {
-		if tagValuesQuery {
+		if strings.Contains(strings.ToLower(query.RawQuery), strings.ToLower("SHOW TAG VALUES")) {
 			if len(valuePair) >= 2 {
 				values = append(values, valuePair[1].(string))
 			}
-		} else if retentionPolicyQuery {
-			// We want to know whether the given retention policy is the default one or not.
-			// If it is default policy then we should add it to the beginning.
-			// The index 4 gives us if that policy is default or not.
-			// https://docs.influxdata.com/influxdb/v1.8/query_language/explore-schema/#show-retention-policies
-			// Only difference is v0.9. In that version we don't receive shardGroupDuration value.
-			// https://archive.docs.influxdata.com/influxdb/v0.9/query_language/schema_exploration/#show-retention-policies
-			// Since it is always the last value we will check that last value always.
-			if len(valuePair) >= 1 {
-				if valuePair[len(row.Columns)-1].(bool) {
-					values[0] = valuePair[0].(string)
-				} else {
-					values = append(values, valuePair[0].(string))
-				}
-			}
```

Comment:
Because of this https://github.com/grafana/grafana/pull/72467 we don't need to find and return the default one anymore.